packages/wonder-blocks-forms/components/dropdown.js

```diff
@@ -0,0 +1,121 @@
+// @flow
+import * as React from "react";
+import {StyleSheet} from "aphrodite";
+
+import {View} from "wonder-blocks-core";
+
+type Props = {
+    onClose: () => void,
+    onOpen: () => void,
+    onChange: (e: SyntheticEvent<>, value: string) => void,
+    show: boolean,
+    selection: ?any,
+    controlled: boolean,
+    items?: Array<{
+        label: string,
+        value: any,
+    }>,
+};
+
+type State = {
+    show: boolean,
+    selection: ?any,
+};
+
+export default class Dropdown extends React.Component<Props, State> {
+    static defaultProps = {
+        onClose: () => {},
+        onOpen: () => {},
+        onChange: (e: SyntheticEvent<>, value: string) => {},
+        show: false,
+        controlled: true,
+        selection: null,
+    };
+
+    constructor(props: Props) {
+        super(props);
+        this.state = {
+            show: props.show,
+            selection: null,
+        };
+    }
+
+    setState(props: any, callback?: any) {
+        if (this.props.controlled) {
+            super.setState(props, callback);
+        }
+    }
+
+    handleHeaderClick = () => {
+        const show = !this.state.show;
+        this.setState({show});
+        if (show) {
+            this.props.onOpen();
+        } else {
+            this.props.onClose();
+        }
+    };
+
+    handleChange = (e: SyntheticEvent<>, value: any) => {
```

Comment:
I'm wondering if we should always include the original event in callbacks just in case people need to do something with it like with buttons and confetti.